vic/drivers/cesm/src/cesm_put_data.c

```diff
@@ -0,0 +1,295 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * Collect coupled fields and export to coupler
+ *
+ * Sign convention: positive value <=> downward flux
+ * Units: see notes in code or seq_flds_mod.F90
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2014 The Land Surface Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <vic_driver_cesm.h>
+
+void
+vic_cesm_put_data()
+{
```

Comment:
Just out of curiosity, why duplicate the regular put_data() function logic here?  Why not assign the outputs of put_data(), already aggregated over veg and band, to the l2x variables?  That way, any changes in how variables are aggregated can be made in one place instead of two.  For example, the lake and treeline stuff below would not be an issue here if it were done that way.